src/perf.js

```diff
@@ -0,0 +1,12 @@
+export const memoize = function (fn) {
```

Comment:
Can you paste a code comment with a reference / links to where we're getting our memoize function from? There are a ton of battle-tested examples out there, and hopefully we can leverage a good, well-trodden path here.